### DISCUSSION TITLE
Library behaviour for sametimeresponses

### DISCUSSION THREAD

**Discussion by vladstarikov88** [Flyrell/axios-auth-refresh#189] [Q&A]
Hello!

I'm interesting about case I faced into.

Imagine situation when token was expired and user is trying to call some requests. For example he loads `Dashboard` page with async requests `/books`, `/games`. And of course each of this responses will return `401` code.

And we have to get new token pair only once. Since it makes no sense to request a new pair of tokens twice

Does your library meet this requirement?

---------------------------

I've already solved the problem locally, only now I'm stuck on how to now return responses to the original requesters using the new token. So I want to use your solution

**Comment by tibbe** [comment]
Having the same quest. I tried to use `onRetry` to add the refreshed token to all (concurrent) requests but it didn't get called at all.
